File: solutions/3099. Harshad Number/3099.cpp
class Solution {
 public:
  int sumOfTheDigitsOfHarshadNumber(int x) {
    const int digitSum = getDigitSum(x);
    return x % digitSum == 0 ? digitSum : -1;
  }

 private:
  int getDigitSum(int x) {
    int digitSum = 0;
    while (x > 0) {
      digitSum += x % 10;
      x /= 10;
    }
    return digitSum;
  }
};

File: solutions/3099. Harshad Number/3099.py
class Solution:
  def sumOfTheDigitsOfHarshadNumber(self, x: int) -> int:
    digitSum = self._getDigitSum(x)
    return digitSum if x % digitSum == 0 else -1

  def _getDigitSum(self, num: int) -> int:
    return sum(int(digit) for digit in str(num))
